### PR TITLE
Avoid using register x8 as it will be used as the RetBuf argument in the future

### PR DESCRIPTION
**Pull request by briansull**
Placed usage of REG_SPILL_CHOICE under LEGACY_BACKEND and removed them for RyuJit only targets
Moved REG_R8 for ARM64 to later in the REG_VAR_ORDER
Change the definition of REG_PINVOKE_ registers to avoid using REG_R8
Change the definition of REG_TMP_0 and REG_TMP_1 to avoid using REG_R8
 the REG_TMP defines are only used in one place: CodeGen::genEmitGSCookieCheck

### CODE REVIEW

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Comment by briansull** [comment]
@kyulee1 @sivarv @CarolEidt PTAL
/cc: @dotnet/jit-contrib

**Comment by CarolEidt** [comment]
LGTM

File: src/jit/target.h [hunk 1]
```diff
@@ -1529,8 +1530,8 @@ typedef unsigned short          regPairNoSmall; // arm: need 12 bits
   #define RBM_ALLFLOAT            (RBM_FLT_CALLEE_SAVED | RBM_FLT_CALLEE_TRASH)
   #define RBM_ALLDOUBLE            RBM_ALLFLOAT
 
-  #define REG_VAR_ORDER            REG_R8,REG_R9,REG_R10,REG_R11,REG_R12,REG_R13,REG_R14,REG_R15,\
-                                   REG_R7,REG_R6,REG_R5,REG_R4,REG_R3,REG_R2,REG_R1,REG_R0,\
+  #define REG_VAR_ORDER            REG_R9,REG_R10,REG_R11,REG_R12,REG_R13,REG_R14,REG_R15,\
```
Comment (sivarv) [inline src/jit/target.h]: It might be a good idea to add a comment here why R8 is not first in the list though a callee saved reg.

**Comment by sivarv** [comment]
Looks good.

File: tests/arm64/Tests.lst [hunk 2]
```diff
@@ -35795,7 +35795,7 @@ RelativePath=JIT\jit64\localloc\ehverify\eh11_large\eh11_large.cmd
 WorkingDir=JIT\jit64\localloc\ehverify\eh11_large
 Expected=0
 MaxAllowedDurationSeconds=600
-Categories=Pri0;EXPECTED_FAIL;REGRESS
+Categories=Pri0;EXPECTED_PASS
```
Comment (kyulee1) [inline tests/arm64/Tests.lst]: This test often fails in the release, that's why i put it in fail before.

File: src/jit/target.h (hunk 1 above)
Comment (briansull) [inline src/jit/target.h]: I will be adding a new define that refers to REG_R8 as the REG_RET_BUFF
That should clear things up about why R8 is located where it is.

**Comment by kyulee1** [comment]
LGTM

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Pull request merged**